File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        sort(strs.begin(),strs.end());
        
        string s1=strs[0];
        string s2=strs[strs.size()-1];
        
        int i=0;
        while(i<s1.length())
        {
            if(s1[i]==s2[i])
                i++;
            
            else
                break;
        }
        return i==0?"":s1.substr(0,i);
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        
        
        strs.sort()
        
        n=len(strs)
        
        str1=strs[0]
        str2=strs[n-1]
        
        if str1=="":
            return ""
        
       
        
        
        for idx in range(0,len(str1)):
            
            if str1[idx]==str2[idx]:
                idx+=1
                
            else:
                break
                
        return str1[:idx]
        
        
                
                
                
                
                
                
       
    
            
            
        
        
       
            
            
            
        